lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -110,7 +123,8 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
           raise(ArgumentError, "with_puppet_running_on's conf_opts must be a Hash. You provided a #{conf_opts.class}: '#{conf_opts}'") if !conf_opts.kind_of?(Hash)
           cmdline_args = conf_opts[:__commandline_args__]
           service_args = conf_opts[:__service_args__] || {}
-          conf_opts = conf_opts.reject { |k,v| [:__commandline_args__, :__service_args__].include?(k) }
+          restart_when_done = conf_opts.fetch(:restart_when_done, false)
```

Comment:
:+1: to preserving beaker's current behavior by default. That said I would like to not have to change all 59 foss tests that use with_puppet_running_on. Can we default to true, merge whatever value is set on the `host`, and merge whatever value is passed as a parameter. That way the last one takes precedence?